include/zephyr/spinlock.h

```diff
@@ -29,6 +29,56 @@ extern "C" {
  * @{
  */
 
+/**
+ * @brief Leaves a code block guarded with @ref K_SPINLOCK_GUARDED_SECTION after
+ * releasing the spinlock.
+ *
+ * See @ref K_SPINLOCK_GUARDED_SECTION for details.
+ */
+#define K_SPINLOCK_LEAVE_GUARDED_SECTION continue
```

Comment:
Maybe just "K_SPINLOCK_BREAK", since that's what it's doing?  Not a big fan of long jargon terms like "GUARDED_SECTION" personally.